code/Controllers/HomeController.cs

```diff
@@ -45,17 +46,42 @@ public HomeController(BagsApi bagsApi)
 				if (product?.Images != null && product.Images.Count() > 0)
 					model.Images = product.Images
 						.OrderBy(image => image.Priority)
-						.SelectAndSwallowReferenceType(image =>new Uri(image.Large))
+						.SelectAndSwallowReferenceType(image => new Uri(image.Large))
 						.Where(image => image != null);
 
 				if (product?.Tags != null && product.Tags.Count() > 0)
 				{
 					var brand = product.Tags.FirstOrDefault(x => x.IsBrand)?.TagName ?? "";
-					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x=> x.TagName).ToArray());
+					var styles = String.Join("/", product.Tags.Where(x => x.IsStyle).Select(x => x.TagName).ToArray());
 
 					model.Title = $"{brand} {product.Name} - {styles}";
 				}
 			}
+			else if (!String.IsNullOrEmpty(tagId))
+			{
+				model.Url = $"https://bagcupid.com/app/{path.TrimStart('/')}";
+
+				var tags = tagId.Split('_');
+				var products = await bagsApi.GetProductsByTags(tags);
+
+				if (products?.Count > 0)
+					model.Images = products.Take(5)
+						.Select(product => product.Images
+							.OrderBy(image => image.Priority)
+							.SelectAndSwallowReferenceType(image => new Uri(image.Large))
+							.Where(image => image != null).FirstOrDefault());
+
+
+				var tagModels = await bagsApi.GetTags();
+
+				if (tagModels?.Count() > 0)
+				{
+					var tagsDescription = String.Join(", ", tagModels.Where(x => tags.Any(y => Convert.ToUInt32(y) == x.TagId))
+						.Select(x => x.TagName));
```

Comment:
Style: Multiply nested LINQ statements can become really hard to read.  This one is particularly tricky so I recommend breaking it up into several separate steps with well named variables in-between.  Alternatively, it may be easier to just put all of the tags into a dictionary and then do a lookup by ID.

Readability: Use more verbose variable names (not `x` and `y`).

```c#
var tags = tagModels.ToDictionary(tagModel => tagModel.TagId, tagModel => tagModel.TagName);
var tagNames = tags.Select(tagId => tags[Convert.ToUInt32(tagId)]);
var tagsDescription = String.Join(", ", tagNames);
```

The first line extracts all of the tags into a dictionary for easy lookup.  The second line gets an enumerable of tag names from the IDs coming in from the query string.  The third line joins them all together.